realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/RealmFunctionsTests.kt

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import androidx.test.platform.app.InstrumentationRegistry
+import org.bson.*
+import org.bson.codecs.StringCodec
+import org.bson.codecs.configuration.CodecRegistries
+import org.bson.codecs.pojo.PojoCodecProvider
+import org.bson.types.Decimal128
+import org.bson.types.ObjectId
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import kotlin.test.assertEquals
+
+class RealmFunctionsTests {
+
+    private lateinit var app: TestRealmApp
+    private lateinit var functions : RealmFunctions
+
+    @Before
+    fun setup() {
+        Realm.init(InstrumentationRegistry.getInstrumentation().targetContext)
+        app = TestRealmApp()
+        functions = RealmFunctions(app.configuration.codecRegistry)
```

Comment:
The functions should be created independently but come from them the user `user.getFunctions()` which I guess becomes `user.functions` in Kotlin. We should make the RealmFunctions constructor package-private if possible.

Reply:
The functions need to be tied to a user. This isn't happening in ObjectStore right now, so we need to fix it there as well and pass in the user as a constructor argument from Java

Reply:
Looking at the RealmFunctions class it was marked as "in-progress" I guess this is fine for now.

Reply:
Yep. All actual function-implementation (creating it from users, etc.) is in #6810, as it requires OS bson stuff, and I did not want to pull that in yet.